modules/extensions-mule-support/src/test/java/org/mule/runtime/module/extension/mule/internal/loader/ast/AbstractMuleSdkAstTestCase.java

```diff
@@ -45,13 +45,13 @@ public abstract class AbstractMuleSdkAstTestCase extends AbstractMuleTestCase {
 
   private final Map<String, String> properties = new HashMap<>();
 
-  private static List<ExtensionModel> runtimeExtensionModels;
+  protected static Set<ExtensionModel> runtimeExtensionModels;
 
   protected abstract String getConfigFile();
 
   @BeforeClass
   public static void beforeClass() throws Exception {
-    runtimeExtensionModels = new ArrayList<>();
+    runtimeExtensionModels = new HashSet<>();
```

Comment:
why? this might turn the test flaky. At most, it should be a LinkedHashSet to maintain behavior

Reply:
The reason I changed it to a `Set` impl was because in all places we were using it, we either required a `Collection` or a `Set` ([here](https://github.com/mulesoft/mule/blob/6b0eee046bc371e81193208c788a287be85d9203/modules/extensions-mule-support/src/test/java/org/mule/runtime/module/extension/mule/internal/loader/MuleSdkPluginExtensionModelLoaderTestCase.java#L169)), and for the later, we would have to be creating the `Set` from the `List` anyway everytime.

Changed to `LinkedHashSet`.